src/library/scala/Array.scala

```diff
@@ -188,8 +188,11 @@ object Array {
   // Array(e0, ..., en) is translated to { val a = new Array(3); a(i) = ei; a }
   def apply[T: ClassTag](xs: T*): Array[T] = {
     val array = new Array[T](xs.length)
+    val iterator = xs.iterator
     var i = 0
-    for (x <- xs.iterator) { array(i) = x; i += 1 }
+    while (iterator.hasNext) {
+      array(i) = iterator.next(); i += 1
+    }
```

Comment:
These are always expanded in `cleanup` if they're called with individual args, right? Could this just delegate to `toArray` or `copyToArray` or whatnot, if being called with a splat?

Reply:
That sounds like a worthwhile improvement, but would take some care to make sure it has the right characteristics in all implementations of whichever method we delegate to. I'd like to keep using iterators here and just focus on avoiding the `IntRef`.